docs/DeveloperGuide.md

```diff
@@ -1,38 +1,235 @@
+1. Command, Parser and UI java files are adapted from one of our group member's [Duke Project](https://github.com/MingEn82/tp)
+---
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+---
+### Architecture
 
+![](../out/docs/uml/Architecture/architecture.svg)
 
-## Product scope
-### Target user profile
+**Main Components of LifeTracker**
+
+`LifeTracker` is where users will access the application from. It is responsible for:
+
+- At app launch: Initialise the Databases and UI classes
+- At app closure: Saves all updated data to databases
+
+The rest of the Application consists of three components:
+
+- `UI`: Reponsible for I/O between user and application
+- `Command`: Handles execution of user inputs
+- `Storage`: Stores information of the user and meals eaten
+
+---
+### UI Component
+The UI is made up of three classes, `GeneralUi`, `CalorieUi` and `WeightUi`, and together,
+they improve on the accessibility of the application.
+
+The `UI` component,
+- Displays what are the possible inputs at any one time
+- Allows the user to check that a value added is correct
+- Informs the user whenever there is an invalid input
+
+### Storage Component
+
+![storage-class-diagram](./uml/storage-class-diagram.png)
+
+The `Storage` is the base class which all `Storage` components inherit from. There are currently four storages:
+
+1. `FoodStorage`: For parsing and storing all of the foods supported by _LifeTracker_
+2. `UserStorage`: For storing the user's profile
+3. `MealStorage`: For storing the meals that the user ate
+4. `ExerciseStorage`: For storing the exercises that the user did
+
+#### Interfaces
+
+To ensure that each `Storage` component only implements methods for writing and reading when needed, we created two interfaces
+
+1. `FileReadable`: Has `load()` method for `Storage` component to be able to read from a database
+2. `FileWritable`: Has `write()` method for `Storage` component to be able to write to a database
+
+For example, `FoodStorage` is not meant to be edited by user, hence it does not implement `FileWritable`
+
+# Implementation
+This section describes some noteworthy details on how certain features are implemented.  
+<p>&nbsp;</p>
+
+## [Proposed] Add meal feature
+
+### Proposed Implementation
+
+The proposed mechanism for adding a meal is facilitated by `AddMealCommand`. It extends `Command` and overrides the
+`execute` method in the `Command` class.
+
+In this command, there are 2 ways for the user to add a meal to storage. 
+1. Using a one-line command in their CLI in the format `add [DATE] /on [MEAL_TYPE] /type [FOOD_INDEX] /foods`
+2. Typing `add` into the CLI and following the printed prompts to enter the date in the specified format, type of meal and specific food.
+
+<img src="uml/AddMealCommand1.png" alt="Sequence Diagram" width="500">
```

Comment:
Arrow from User to :AddMealCommand is missing a name for method call which is execute().